Modules/Loadable/Markups/Logic/vtkSlicerMarkupsLogic.h

```diff
@@ -243,6 +245,35 @@ class VTK_SLICER_MARKUPS_MODULE_LOGIC_EXPORT vtkSlicerMarkupsLogic :
   void RegisterJsonStorageNodeForMarkupsType(std::string markupsType, std::string storageNodeClassName);
   vtkMRMLMarkupsJsonStorageNode* AddNewJsonStorageNodeForMarkupsType(std::string markupsType);
 
+  /// Registers a markup and its corresponding widget to be handled by the Markups module
+  /// For a markup to be handled by this module (processed by the displayable
+  /// manager, UI and subject hierarchy) it needs to be registered using this function.
+  /// \param markupsNode MRMLMarkups node to be registered.
+  /// \param markupsWidget vtkSlicerWidget associated to the MRMLMarkups node registered.
+  void SetMarkup(vtkMRMLMarkupsNode *markupsNode, vtkSlicerMarkupsWidget* markupsWidget);
+
+  /// This returns an isntance to a corresponding vtkSlicerMarkupsWidget associated
+  /// to the indicated markups name.
+  /// \param nodeClass registered class to retrieve the associated widget.
+  /// \return pointer to associated vtkSLicerMarkupsWidget or nullptr if the MRML node
+  /// class is not registered.
+  vtkSlicerMarkupsWidget* GetWidgetByMarkupsName(const char* markupName) const;
+
+  /// This returns an isntance to a corresponding vtkMRMLMarkupsNode associated
+  /// to the indicated markups name.
+  /// \param nodeClass registered class to retrieve the associated widget.
+  /// \return pointer to associated vtkSLicerMarkupsWidget or nullptr if the MRML node
+  /// class is not registered.
+  vtkMRMLMarkupsNode* GetNodeByMarkupsName(const char* markupName) const;
+
+  /// This returns a constant reference to the markups-widgets association map.
+  const std::map<std::string, vtkSmartPointer<vtkSlicerMarkupsWidget> >& GetMarkupsWidgets()const
```

Comment:
It would be better to avoid such low-level access, unless it is unavoidable. I think it is not Python-wrappable either (could you check?).

Reply:
Yes, this can be avoided. I've made the changes accordingly.